src/cursor/cur_backup_incr.c

```diff
@@ -9,53 +9,75 @@
 #include "wt_internal.h"
 
 /*
- * __alloc_merge --
- *     Merge two allocation lists.
+ * __wt_backup_load_incr --
+ *     Load the incremental.
  */
-static void
-__alloc_merge(
-  uint64_t *a, uint64_t a_cnt, uint64_t *b, uint64_t b_cnt, uint64_t *res, uint64_t *res_cnt)
+int
+__wt_backup_load_incr(
+  WT_SESSION_IMPL *session, WT_CONFIG_ITEM *blkcfg, WT_ITEM *bitstring, uint64_t nbits)
 {
-    uint64_t total;
-
-    for (total = 0; a_cnt > 0 || b_cnt > 0; ++total, res += 2) {
-        if (a_cnt > 0 && b_cnt > 0) {
-            if (a[0] <= b[0]) {
-                res[0] = a[0];
-                if (a[0] + a[1] < b[0])
-                    res[1] = a[1];
-                else {
-                    res[1] = (b[0] + b[1]) - a[0];
-                    b += 2;
-                    --b_cnt;
-                }
-                a += 2;
-                --a_cnt;
-            } else if (b[0] <= a[0]) {
-                res[0] = b[0];
-                if (b[0] + b[1] < a[0])
-                    res[1] = b[1];
-                else {
-                    res[1] = (a[0] + a[1]) - b[0];
-                    a += 2;
-                    --a_cnt;
-                }
-                b += 2;
-                --b_cnt;
-            }
-        } else if (a_cnt > 0) {
-            res[0] = a[0];
-            res[1] = a[1];
-            a += 2;
-            --a_cnt;
-        } else if (b_cnt > 0) {
-            res[0] = b[0];
-            res[1] = b[1];
-            b += 2;
-            --b_cnt;
+    if (blkcfg->len != 0)
+        WT_RET(__wt_nhex_to_raw(session, blkcfg->str, blkcfg->len, bitstring));
+    if (bitstring->size != (nbits >> 3))
+        goto format;
+    return (0);
+format:
+    WT_RET_MSG(session, WT_ERROR, "corrupted modified block list");
```

Comment:
Since there's only one use of the `goto format`, I'd just inline the `WT_REF_MSG`, you don't need a goto.
```
*** cur_backup_incr.c   Tue Feb  4 20:12:51 2020
--- -   Tue Feb  4 20:24:01 2020 
***************  
*** 19,28 ****   
      if (blkcfg->len != 0)
          WT_RET(__wt_nhex_to_raw(session, blkcfg->str, blkcfg->len, bitstring));
      if (bitstring->size != (nbits >> 3)) 
!         goto format;
      return (0);
- format:
-     WT_RET_MSG(session, WT_ERROR, "corrupted modified block list");
  }   
      
  /*  
--- 19,26 ----
      if (blkcfg->len != 0)
          WT_RET(__wt_nhex_to_raw(session, blkcfg->str, blkcfg->len, bitstring));
      if (bitstring->size != (nbits >> 3)) 
!       WT_RET_MSG(session, WT_ERROR, "corrupted modified block list");
      return (0);
  }   
      
  /*  
```

Reply:
Fixed.